tenseal/utils/utils.h

```diff
@@ -0,0 +1,52 @@
+#ifndef TENSEAL_UTILS_UTILS_H
+#define TENSEAL_UTILS_UTILS_H
+
+#include <seal/seal.h>
+
+#include <memory>
+
+namespace tenseal {
+
+using namespace seal;
+using namespace std;
+
+/*
+Replicate the current vector as many times to fill `final_size` elements.
+*/
+template <typename T>
+void replicate_vector(vector<T>& vec, size_t final_size) {
+    if (vec.empty()) {
+        throw invalid_argument("can't replicate an empty vector");
+    }
+    size_t init_size = vec.size();
+    vec.reserve(final_size);
+    for (size_t i = 0; i < final_size - init_size; i++) {
+        vec.push_back(vec[i % init_size]);
+    }
+}
+
+/*
+Apply modulus switching to the ciphertext (or plaintext) having the higher
+modulus.
+*/
+template <typename T>
+void set_to_same_mod(shared_ptr<TenSEALContext> context, Ciphertext& ct,
+                     T& other) {
+    size_t idx1 =
+        context->seal_context()->get_context_data(ct.parms_id())->chain_index();
+    size_t idx2 = context->seal_context()
```

Comment:
Might be easier with a lambda function. Something like

```
auto get_chain_index = [&] (const auto& obj) -> std::optional<size_t>{
  auto ctx_data = context->seal_context()->get_context_data(obj.parms_id());
  if (ctx_data == nullptr) {
    return {}; 
  }
  return ctx_data->chain_index();
};

size_t idx1 = get_chain_index(ct).value_or(/*other value ?*/);
size_t idx2 = get_chain_index(other).value_or(/*other value ?*/);
```


(sorry, I haven't tried to compile, but it should work - it's just a suggestion).
Here, I've added a lambda function that returns an optional chain_index.
``` [&] ``` - means it captures the current context, you can access the local variables.
```optional<size_t>``` - means it can hold a size_t value, otherwise it won't use the memory.

Reply:
the question is what value should we return here, couldn't figure it out yet

Reply:
since we are relying on SEAL exceptions anyway, we could throw an exception here too. 
For example, instead of returning {} in the lambda function. And optional is not needed there, in this scenario.